src/components/CV_hierarchy/hierarchy.js

```diff
@@ -167,66 +169,120 @@ function CustomizedTreeView(props) {
+      const updateMaps = (currSchema) => {
+        for (let key in currSchema) {
+          newIdToLabelMap.set(key, currSchema[key]["label"]["en"]);
+          newLabelToIdMap.set(currSchema[key]["label"]["en"], key);
+
+          for (const childSchema of currSchema[key]["children"]) {
+            updateMaps(childSchema); // Recursively update maps
+          }
+        }
+      };
+
+      // Call the recursive function to update maps
+      updateMaps(schema);
+
+      // Update state after the recursion is completed
+      setIdToLabelMap(newIdToLabelMap);
+      setLabelToIdMap(newLabelToIdMap);
+    }
     const path = Array.from(new Set(value.map(v => findPath(schema, v)).flat()));
-    setExpanded(prevExpaned => path.length > prevExpaned.length ? path : prevExpaned)
+    setExpandedItems(prevExpaned => path.length !== prevExpaned.length ? parseLabelArrayToIdArray(path, labelToIdMap) : prevExpaned)
     // calculate the counts 
     if (Array.isArray(facetValues)){
       setCountMap(new Map()); // initialize counts 
       calculateCounts(schema);
     }
-    setSelected(value);
-  }, [schema, value, facetValues, calculateCounts])
+    setSelectedItems(parseLabelArrayToIdArray(value, labelToIdMap));
+  }, [schema, value, facetValues, calculateCounts, labelToIdMap, idToLabelMap])
 
-  const handleToggle = (event, nodeIds) => {
-    const difference = nodeIds
-      .filter(x => !expanded.includes(x))
-      .concat(expanded.filter(x => !nodeIds.includes(x)));
+  const handleToggle = (event, itemIds) => {
+    const difference = itemIds
+      .filter(x => !expandedItems.includes(x))
+      .concat(expandedItems.filter(x => !itemIds.includes(x)));
     // For toggle items, we could use ctrl + enter to select the tree item
     if (event.ctrlKey && event.code === 'Enter') {
-      onClick(difference[0]);
+      onClick(parseIdArrayToLabelArray(difference[0], idToLabelMap), "add");
     } else {
-      setExpanded(nodeIds);
+        setExpandedItems(itemIds);
     }
   };
 
-  const handleSelect = (event, nodeIds) => {
-    // nodeIds[0] is the selected label 
-    if (value.includes(nodeIds[0])){
+  const handleSelect = (event, itemIds) => {
+    // itemIds[0] is the selected label 
+    let labelIds = parseIdArrayToLabelArray(itemIds, idToLabelMap)
+    if (value.includes(labelIds[0])){
       // remove the selected label
-      onClick(nodeIds[0], "delete");
+      onClick(labelIds[0], "delete");
     } else{
       // add the selected label
-      onClick(nodeIds[0], "add" )
+      onClick(labelIds[0], "add" )
     }
   };
 
   const handleFilter = (event) => {
     const { value } = event.target;
     setFilter(value);
     if (value.trim().length === 0) {
-      setExpanded([firstLevel]);
+      setExpandedItems(parseLabelArrayToIdArray([firstLevel], labelToIdMap));
     } else {
-      setExpanded(findPath(schema, value));
+      setExpandedItems(parseLabelArrayToIdArray(findPath(schema, value), labelToIdMap));
```

Comment:
Is `setExpanded` vs. `setExpandedItems` due to the API update?

Reply:
Yes.